File: codeforces_1490C.py
import math

qubes = set(i*i*i for i in range(1, 10000))

for _ in range(int(input())):
    x = int(input())
    top = int(x**(1/3)) + 1

    for a in range(1, top):
        if (x - a*a*a) in qubes:
            print("YES")
            break
    else:
        print("NO")

File: codeforces_1490C.cpp
#include <iostream>
#include <cmath>

using namespace std;
int main() {
	int t;
	cin >> t;
	while (t--) {
		long long x;
		cin >> x;
		long long top = std::cbrt(x)+1;

		bool ok = false;
		for (long long a = 1; a < top; a++) {
			long long lo = 1;
			long long hi = top;
			long long a3 = std::pow(a, 3);

			while (lo + 1 < hi) {
				long long mid = (lo + hi)/2;
				if (std::pow(mid, 3) <= x - a3) {
       					lo = mid;
				} else {
					hi = mid;
				}
			}

			if (a3 + std::pow(lo, 3) == x) {
				cout << "YES\n";
				ok = true;
				break;
			}
		}

		if (!ok) {
			cout << "NO\n";
		}
	}
}
